source/monitoring/logging.rst

```diff
@@ -0,0 +1,105 @@
+
+For logs relating to issues about: 
+
+ - Authenticating to the web-node
+ - Configuring OnDemand 
+
+Look in either of:
+
+:file:`/var/log/httpd/<hostname>_error.log`
+
+- OnDemand will log to this location for its own configuration errors.
+
+:file:`/var/log/httpd/<hostname>_access.log`
+
+- Where OnDemand will log succsseful logins.
+
+.. warning::
+
+    There are no entries for failed logins.
+
+**NGINX Logs**
+
+The NGINX logs are the output of the user :ref:`PUN <glossary>`. These logs will capture things relevant 
+to a *particular user* such as:
+
+- Debuggin issues related to job submissions for a user. For example, commands being issued to the scheduler 
+  (``sbatch``, ``qsub``, etc) can be seen here by searching for ``execve``.
+- Issues related to PUNs crashing and/or pages not rendering correctly.
+
+These logs are located at:
+
+:file:`/var/log/ondemand-nginx/<user>`
+
+.. note::
+
+    These logs by default have ``root:root`` ownership. It may be beneficial 
+    to ``chown`` an appropriate staff group on these files for troubleshooting down the road.
+
+Session Data
+------------
+These files will provide information around connections between the user's PUN and a 
+:ref:`compute-node <glossary>`. 
+
+These logs provide *information around connections* and are also *the working directory of the job and location 
+of* ``stderr`` *and* ``stdout`` for a job (typically to ``output.log``). Other apps will also output 
+information they may need there as well for connections and errors.
+
+
+.. note::
+
+    The files will be *owned by the user* and so admins will need to ensure they are either 
+    able to substitue user or escalate to root in order to see these files.
+
+In general the session data or job submission files for apps across the dashboard, such as the Job Composer, 
+Batch Connect, or Frame-renderer, all start from the root of:
+
+:file:`~/ondemand/data/sys/`
+
+**Batch Connect Apps**
+
+User session data for batch connect apps can be seen from a *user's* home directory at:
+
+:file:`~/ondemand/data/sys/dashboard/batch_connect/sys/<app>/output/<session id>`
+
+This file is used for the session data presented on the interactive apps page and data 
+used to connect to the batch connect app.
+
+.. note::
+
+    There may be more than one file in this directory, and sometimes you will have to look through 
+    various files to find the correct session data being used by looking at things like the ``created_at`` 
+    field in the file.
+
+**Example**
+
+Suppose a user is having trouble connecting to a Codeserver session they created. 
+To see what data is being used by this batch connect app for the connection, look in:
+
+.. code-block:: sh
+    
+    cat ~/ondemand/data/sys/dashboard/batch_connect/sys/<app>/output/<session id>
+    
+This will output a json object which will give the information to debug 
+any issues this session is having when connecting, such as ``id``, ``token``, ``cluster_id``, 
+and ``job_id``.
```

Comment:
I think the example here is still around looking at the db file?  I think `output.log` here is pretty reliable. Like every BC app is going to have the scheduler redirect stdout and stderror to the `<%= staged_root %>/output.log` unless they override it which I'd say is a rare edge case - or at least a change made by an informed user (so not the reader of this document).

Reply:
Yes thank you for catching that. I updated the example and tried to include some useful information as to what that `output.log` would capture which the admin may be interested in for troubleshooting as well.